beacon_node/beacon_chain/src/data_availability_checker.rs

```
@@ -445,8 +445,6 @@ impl<T: BeaconChainTypes> DataAvailabilityChecker<T> {
             .flatten()
             .map(CustodyDataColumn::into_inner)
             .collect::<Vec<_>>();
-        let all_data_columns =
-            RuntimeVariableList::from_vec(all_data_columns, T::EthSpec::number_of_columns());
```

What was this line _trying_ to do? I guess it was assuming these columns were for a single block and coercing them to a `RuntimeVariableList` in order to "check" the length?

---

I think this pattern with the `RuntimeVarList::from_vec` of truncating to the length is a footgun for internal functions (non spec enforcing stuff) because we may be passing a vec with length greater than the `max_len` but the extra items get silently dropped and it breaks stuff elsewhere.

---

Ohh just saw you made an issue 🙏

---

PR here:

- https://github.com/sigp/lighthouse/pull/7930

---

> What was this line _trying_ to do? I guess it was assuming these columns were for a single block and coercing them to a `RuntimeVariableList` in order to "check" the length?

Yeah i think exactly this i think